### ISSUE TITLE
Error : FactionCommands.php on line 934" (EXCEPTION) in "src/pocketmine/Server" at line 864

### ISSUE THREAD

**Issue by teracube** [poggit-orphanage/FactionsPro#63]
https://poggit.pmmp.io/r/47567/FactionsPro_dev-41.phar
https://jenkins.pmmp.io/job/PocketMine-MP/1735/artifact/PocketMine-MP.phar

```
2019-01-04 [21:52:26] [Server thread/CRITICAL]: TypeError: "Argument 1 passed to pocketmine\Server::getPlayerExact() must be of the type string, null given, called in phar:///home/myserver/plugins/FactionsPro_dev-41.phar/src/FactionsPro/FactionCommands.php on line 934" (EXCEPTION) in "src/pocketmine/Server" at line 864
2019-01-04 [21:52:26] [Server thread/DEBUG]: #0 FactionsPro_dev-41.phar/src/FactionsPro/FactionCommands(934): pocketmine\Server->getPlayerExact(NULL )
2019-01-04 [21:52:26] [Server thread/DEBUG]: #1 FactionsPro_dev-41.phar/src/FactionsPro/FactionMain(118): FactionsPro\FactionCommands->onCommand(pocketmine\Player object, pocketmine\command\PluginCommand object, string f, array Array())
2019-01-04 [21:52:26] [Server thread/DEBUG]: #2 src/pocketmine/command/PluginCommand(58): FactionsPro\FactionMain->onCommand(pocketmine\Player object, pocketmine\command\PluginCommand object, string f, array Array())
2019-01-04 [21:52:26] [Server thread/DEBUG]: #3 src/pocketmine/command/SimpleCommandMap(256): pocketmine\command\PluginCommand->execute(pocketmine\Player object, string f, array Array())
2019-01-04 [21:52:26] [Server thread/DEBUG]: #4 src/pocketmine/Server(1960): pocketmine\command\SimpleCommandMap->dispatch(pocketmine\Player object, string f allywith LaCoALiTion)
2019-01-04 [21:52:26] [Server thread/DEBUG]: #5 src/pocketmine/Player(2173): pocketmine\Server->dispatchCommand(pocketmine\Player object, string f allywith LaCoALiTion)
2019-01-04 [21:52:26] [Server thread/DEBUG]: #6 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(229): pocketmine\Player->chat(string /f allywith LaCoALiTion)
2019-01-04 [21:52:26] [Server thread/DEBUG]: #7 src/pocketmine/network/mcpe/protocol/CommandRequestPacket(54): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleCommandRequest(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2019-01-04 [21:52:26] [Server thread/DEBUG]: #8 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(99): pocketmine\network\mcpe\protocol\CommandRequestPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2019-01-04 [21:52:26] [Server thread/DEBUG]: #9 src/pocketmine/network/mcpe/protocol/BatchPacket(114): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2019-01-04 [21:52:26] [Server thread/DEBUG]: #10 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(99): pocketmine\network\mcpe\protocol\BatchPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2019-01-04 [21:52:26] [Server thread/DEBUG]: #11 src/pocketmine/Player(3061): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2019-01-04 [21:52:26] [Server thread/DEBUG]: #12 src/pocketmine/network/mcpe/RakLibInterface(162): pocketmine\Player->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2019-01-04 [21:52:26] [Server thread/DEBUG]: #13 vendor/pocketmine/raklib/src/server/ServerHandler(98): pocketmine\network\mcpe\RakLibInterface->handleEncapsulated(string 170.XXX.XX.13 49687, raklib\protocol\EncapsulatedPacket object, integer 0)
2019-01-04 [21:52:26] [Server thread/DEBUG]: #14 src/pocketmine/network/mcpe/RakLibInterface(102): raklib\server\ServerHandler->handlePacket()
2019-01-04 [21:52:26] [Server thread/DEBUG]: #15 src/pocketmine/network/mcpe/RakLibInterface(92): pocketmine\network\mcpe\RakLibInterface->process()
2019-01-04 [21:52:26] [Server thread/DEBUG]: #16 vendor/pocketmine/snooze/src/SleeperHandler(120): pocketmine\network\mcpe\RakLibInterface->pocketmine\network\mcpe\{closure}()
2019-01-04 [21:52:26] [Server thread/DEBUG]: #17 vendor/pocketmine/snooze/src/SleeperHandler(82): pocketmine\snooze\SleeperHandler->processNotifications()
2019-01-04 [21:52:26] [Server thread/DEBUG]: #18 src/pocketmine/Server(2278): pocketmine\snooze\SleeperHandler->sleepUntil(double 1546635145.9609)
2019-01-04 [21:52:26] [Server thread/DEBUG]: #19 src/pocketmine/Server(2139): pocketmine\Server->tickProcessor()
2019-01-04 [21:52:26] [Server thread/DEBUG]: #20 src/pocketmine/Server(1715): pocketmine\Server->start()
2019-01-04 [21:52:26] [Server thread/DEBUG]: #21 src/pocketmine/PocketMine(249): pocketmine\Server->__construct(BaseClassLoader object, pocketmine\utils\MainLogger object, string /home/myserver/, string /home/myserver/plugins/)
2019-01-04 [21:52:26] [Server thread/DEBUG]: #22 /home/myserver/PocketMine-MP.phar(1): require(string phar:///home/myserver/PocketMine-MP.phar/src/pocketmine/PocketMine.php)
2019-01-04 [21:52:26] [Server thread/INFO]: [21:52:26] Atenayis [-]
2019-01-04 [21:52:26] [Server thread/INFO]: Atenayis[/170.XXX.XX.13:49687] logged out due to Internal server error

```

**Comment by teracube** [comment]
issue with /f allywith

https://poggit.pmmp.io/r/47567/FactionsPro_dev-41.phar
https://jenkins.pmmp.io/job/PocketMine-MP/1737/artifact/PocketMine-MP.phar

```
2019-01-20 [12:25:40] [Server thread/INFO]: [Commandlook] otor du 773> /f allywith drago777
2019-01-20 [12:25:40] [Server thread/CRITICAL]: TypeError: "Argument 1 passed to pocketmine\Server::getPlayerExact() must be of the type string, null given, called in phar:///home/myserver/plugins/FactionsPro_v1.3.14.phar/src/FactionsPro/FactionCommands.php on line 934" (EXCEPTION) in "src/pocketmine/Server" at line 934
2019-01-20 [12:25:40] [Server thread/DEBUG]: #0 plugins/FactionsPro_v1.3.14.phar/src/FactionsPro/FactionCommands(934): pocketmine\Server->getPlayerExact(NULL )
2019-01-20 [12:25:40] [Server thread/DEBUG]: #1 plugins/FactionsPro_v1.3.14.phar/src/FactionsPro/FactionMain(118): FactionsPro\FactionCommands->onCommand(pocketmine\Player object, pocketmine\command\PluginCommand object, string f, array Array())
2019-01-20 [12:25:40] [Server thread/DEBUG]: #2 src/pocketmine/command/PluginCommand(58): FactionsPro\FactionMain->onCommand(pocketmine\Player object, pocketmine\command\PluginCommand object, string f, array Array())
2019-01-20 [12:25:40] [Server thread/DEBUG]: #3 src/pocketmine/command/SimpleCommandMap(261): pocketmine\command\PluginCommand->execute(pocketmine\Player object, string f, array Array())
2019-01-20 [12:25:40] [Server thread/DEBUG]: #4 src/pocketmine/Server(2030): pocketmine\command\SimpleCommandMap->dispatch(pocketmine\Player object, string f allywith drago777)
2019-01-20 [12:25:40] [Server thread/DEBUG]: #5 src/pocketmine/Player(2209): pocketmine\Server->dispatchCommand(pocketmine\Player object, string f allywith drago777)
2019-01-20 [12:25:40] [Server thread/DEBUG]: #6 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(238): pocketmine\Player->chat(string /f allywith drago777)
2019-01-20 [12:25:40] [Server thread/DEBUG]: #7 src/pocketmine/network/mcpe/protocol/CommandRequestPacket(54): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleCommandRequest(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2019-01-20 [12:25:40] [Server thread/DEBUG]: #8 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(108): pocketmine\network\mcpe\protocol\CommandRequestPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2019-01-20 [12:25:40] [Server thread/DEBUG]: #9 src/pocketmine/network/mcpe/protocol/BatchPacket(119): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2019-01-20 [12:25:40] [Server thread/DEBUG]: #10 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(108): pocketmine\network\mcpe\protocol\BatchPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2019-01-20 [12:25:40] [Server thread/DEBUG]: #11 src/pocketmine/Player(3097): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2019-01-20 [12:25:40] [Server thread/DEBUG]: #12 src/pocketmine/network/mcpe/RakLibInterface(170): pocketmine\Player->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2019-01-20 [12:25:40] [Server thread/DEBUG]: #13 vendor/pocketmine/raklib/src/server/ServerHandler(99): pocketmine\network\mcpe\RakLibInterface->handleEncapsulated(string 78.XXX.XXX.98 56516, raklib\protocol\EncapsulatedPacket object, integer 0)
2019-01-20 [12:25:40] [Server thread/DEBUG]: #14 src/pocketmine/network/mcpe/RakLibInterface(110): raklib\server\ServerHandler->handlePacket()
2019-01-20 [12:25:40] [Server thread/DEBUG]: #15 src/pocketmine/network/mcpe/RakLibInterface(100): pocketmine\network\mcpe\RakLibInterface->process()
2019-01-20 [12:25:40] [Server thread/DEBUG]: #16 vendor/pocketmine/snooze/src/SleeperHandler(123): pocketmine\network\mcpe\RakLibInterface->pocketmine\network\mcpe\{closure}()
2019-01-20 [12:25:40] [Server thread/DEBUG]: #17 vendor/pocketmine/snooze/src/SleeperHandler(85): pocketmine\snooze\SleeperHandler->processNotifications()
2019-01-20 [12:25:40] [Server thread/DEBUG]: #18 src/pocketmine/Server(2349): pocketmine\snooze\SleeperHandler->sleepUntil(double 1547983540.5021)
2019-01-20 [12:25:40] [Server thread/DEBUG]: #19 src/pocketmine/Server(2209): pocketmine\Server->tickProcessor()
2019-01-20 [12:25:40] [Server thread/DEBUG]: #20 src/pocketmine/Server(1785): pocketmine\Server->start()
2019-01-20 [12:25:40] [Server thread/DEBUG]: #21 src/pocketmine/PocketMine(249): pocketmine\Server->__construct(BaseClassLoader object, pocketmine\utils\MainLogger object, string /home/myserver/, string /home/myserver/plugins/)
2019-01-20 [12:25:40] [Server thread/DEBUG]: #22 (1): require(string phar:///home/myserver/PocketMine-MP.phar/src/pocketmine/PocketMine.php)
2019-01-20 [12:25:40] [Server thread/INFO]: otor du 773 [-]
2019-01-20 [12:25:40] [Server thread/INFO]: otor du 773[/78.XXX.XXX.98:56516] logged out due to Internal server error
2019-01-20 [12:25:40] [RakLibServer thread/NOTICE]: Blocked 78.XXX.XXX.98 for 5 seconds
```

**Comment by Esch16** [comment]
Yep that's because we should check whether the faction leader is online before actually retrieving the information

**Comment by Awzaw** [comment]
Should be fixed in 8211d2f0475, and would happen when you try to ally with a faction that doesn't exist, or has no leader.
